jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py

```diff
@@ -135,3 +140,24 @@ def test_write_momentmap(cubeviz_helper, spectrum1d_cube, tmp_path):
     plugin._obj.filename = "fake_path/test_file.fits"
     with pytest.raises(ValueError, match="Invalid path"):
         plugin._obj.vue_save_as_fits()
+
+
+@pytest.mark.remote_data
+def test_momentmap_nirspec_prism(cubeviz_helper, tmp_path):
+    uri = "mast:jwst/product/jw02732-o003_t002_nirspec_prism-clear_s3d.fits"
+    download_path = str(tmp_path / Path(uri).name)
+    Observations.download_file(uri, local_path=download_path)
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        cubeviz_helper.load_data(download_path)
+    plugin = cubeviz_helper.plugins['Moment Maps']
+    plugin.calculate_moment()
+    assert isinstance(plugin._obj.moment.wcs, WCS)
+
+    # Because cube axes order is re-arranged by specutils on load, this gets confusing.
+    # There is a swapaxes within Moment Map WCS calculation.
+    sky_moment = plugin._obj.moment.wcs.pixel_to_world(50, 30)
+    sky_cube = cubeviz_helper.app.data_collection[0].meta["_orig_spec"].wcs.celestial.pixel_to_world(30, 50)  # noqa: E501
```

Comment:
Note that Moment Map WCS takes `(x, y)` order but I have to swap that around to get the same sky coordinate from cube. Maybe @rosteen should double check if this is expected behavior given how specutils swap things and then we have to swap it back here and there.

Reply:
Given this line on L66 above:

```python
assert result.shape == (4, 2)  # Cube shape is (2, 2, 4)
```

Moment map dimension is `(ny, nx)` but looks like cube is `(nz, nx, ny)`. So maybe this makes sense? 🤪